### ISSUE TITLE
Improve documentation (docstring)

### ISSUE THREAD

**Issue by nkrot** [fastai/fastai#2975]
Please make an effort adding better docstring to the code. 

For example `help(learner.validate)` outputs only this minimally useful message.

```
validate(ds_idx=1, dl=None, cbs=None) method of fastai.learner.Learner instance
    Validate on `dl` with potential new `cbs`.
```

I receive as output a list of 3 items. What are they?

To put it differently. the above help message does not count as documentation.

**Comment by kanak8278** [comment]
Hey! I would like to add improve the docstrings.
Can you help me out with specifics? :-)

**Comment by tcapelle** [comment]
Fastai's docstring are minimal, one liners. The documentation is on the website or the notebooks where the function is defined. This is intended:
https://docs.fast.ai/learner.html#Learner.validate

**Comment by nkrot** [comment]
> Fastai's docstring are minimal, one liners. The documentation is on the website or the notebooks where the function is defined. This is intended:
> https://docs.fast.ai/learner.html#Learner.validate

@tcapelle , in this case, wouldn't it make sense to put links (URLs) to the documentation in docstrings?

**Comment by tcapelle** [comment]
I agree that there is no easy way to go from the code to the doc. I would also like better docstrings, but don't know how ti could be done without impacting the notebooks and code style.
From someone coming from scikit-learn or pandas/numpy used to read docstring with shift+tab it is very painful to find what you are looking for.

**Comment by satyajitghana** [comment]
> I agree that there is no easy way to go from the code to the doc. I would also like better docstrings, but don't know how ti could be done without impacting the notebooks and code style.
> From someone coming from scikit-learn or pandas/numpy used to read docstring with shift+tab it is very painful to find what you are looking for.
> https://forums.fast.ai/t/better-docstrings/82904

I agree !, even coming from pytorch, where the doc strings have standard style, its gets difficult for me here with single ling strings, when working with something like vscode/pycharm

**Comment by muellerzr** [comment]
> > Fastai's docstring are minimal, one liners. The documentation is on the website or the notebooks where the function is defined. This is intended:
> > https://docs.fast.ai/learner.html#Learner.validate
> 
> @tcapelle , in this case, wouldn't it make sense to put links (URLs) to the documentation in docstrings?

This is already possible. fastai promotes installing `nbdev`, and doing `doc(myfunc)` in nbdev will pull up the documentation, a link to the source code, and a link to it's documentation

**Comment by tcapelle** [comment]
maybe nbdev should be a requirement for fastai then.
```
from nbdev.showdoc import show_doc

show_doc(Learner)
```

**Comment by muellerzr** [comment]
We're not doing that, Jeremy will be working on the doc strings at some point. In the meantime just use nbdev if you're not satisfied with the vanilla. (Literally just do `pip install nbdev` and `doc()` will show the pretty doc)

**Comment by tcapelle** [comment]
Got it, thanks!
![image](https://user-images.githubusercontent.com/18441985/103815004-f60d0e80-5062-11eb-96a2-0fe438eb4534.png)

**Comment by satyajitghana** [comment]
@tcapelle yeah but a shortcut for popping up that would be nice, but maybe this request should go to jupyter notebook . . . although most of the time `shift + tab` brings up the docs

**Comment by kcarnold** [comment]
Unfortunately `doc()` is not the default way of getting docs in a notebook environment. I'm teaching my second class with fastai and I didn't know about it. It would be really helpful if the docstring gave some sort of hint about this, either to also include the docs link, or at least to say something like "see docs(this_function) for more". Perhaps this is something that the nbdev build step could add in.

**Comment by bjornasm** [comment]
> We're not doing that, Jeremy will be working on the doc strings at some point. In the meantime just use nbdev if you're not satisfied with the vanilla. (Literally just do `pip install nbdev` and `doc()` will show the pretty doc)

Any progress on this?
